dropwizard-hibernate/src/main/java/io/dropwizard/hibernate/UnitOfWorkAspect.java

```diff
@@ -178,6 +178,7 @@ private void commitTransaction(UnitOfWork unitOfWork, Session session) {
     }
 
     private static void setContext(UnitOfWork unitOfWork, Session session) {
+        clearContext();
```

Comment:
always clear the context we are about to reset

Reply:
Looks safe, dug into the internals of clearContext a bit and didn't see anything that should throw an exception when there is nothing to clear.